src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -415,6 +415,13 @@ private JsValue AppendTimeSeries(JsValue document, JsValue name, JsValue[] args)
                         throw new ArgumentException($"{signature}: The values should be an array but got {GetTypes(valuesArg)}");
                     }
 
+                    bool timeseriesExists = false;
+                    if (DebugMode)
+                    {
+                        var reader = _database.DocumentsStorage.TimeSeriesStorage.GetReader(_docsCtx, id, timeseries, DateTime.MinValue, DateTime.MaxValue);
```

Comment:
what if we add value to time series which contains many element? Will it send all of them to browser?

Reply:
oh actually we read this to detect if we have any item. Can we check this in more effective way?

Reply:
@karmeli87  
Can this method ` _database.DocumentsStorage.TimeSeriesStorage.AppendTimestamp`
be modified so that it will inform if timeSeries existed and appended -or- whether a new ts was created ?
similar to  `_database.DocumentsStorage.CountersStorage.IncrementCounter`

Reply:
We can, please open an issue for that.
On a brief look on this PR, it seems that running stuff in debug mode should change the actual time-series, right?

Reply:
I would like to have a test for that

Reply:
issue created: https://issues.hibernatingrhinos.com/issue/RavenDB-15367

Reply:
This is only for Debug mode of the patch testing. It's not a hot path so I don't think we need to do anything dedicated here. This is very rare scenario.

Reply:
@karmeli87 
In Debug mode we _don't_ commit the transaction so it won't modify anything.